FileAnalyzer.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Multi_threaded_File_Processing;
+
+internal class FileAnalyzer
+{
+    public readonly string filesDirectory = "InputFiles";
+
+    public FileAnalyzer()
+    {
+        if (!Directory.Exists(filesDirectory))
+        {
+            throw new DirectoryNotFoundException("Directory with files doesn't exist!");
+        }
+    }
+
+    public void CountFiles(object? resetEvent)
+    {
+        if (resetEvent is not ManualResetEvent manualResetEvent)
+        {
+            throw new TypeAccessException($"ManualResetEvent type expected. {resetEvent.GetType()} is given.");
+        }
+        string[] files = Directory.GetFiles(filesDirectory);
+        Console.WriteLine("Count of files is - {0}", files.Length);
+
+        manualResetEvent.Set();
+    }
+
+    public void CountLines(object? resetEvent)
+    {
+        if (resetEvent is not ManualResetEvent manualResetEvent)
+        {
+            throw new TypeAccessException($"ManualResetEvent type expected. {resetEvent.GetType()} is given.");
+        }
+
+        string[] files = Directory.GetFiles(filesDirectory);
+
+        int lineCounter = 0;
+
+        foreach (string file in files)
+        {
+            StreamReader sr = new StreamReader(file);
```

Comment:
Better use using keyword with disposable objects like StreamReader to maintain a common approach